test/Microsoft.AspNet.Mvc.Razor.Host.Test/MvcRazorParserTest.cs

```diff
@@ -15,30 +16,132 @@ namespace Microsoft.AspNet.Mvc.Razor
 {
     public class MvcRazorCodeParserTest
```

Comment:
don't you have tests somewhere that look at invalid combinations e.g. multiple `@model` directives in either a view or a _GlobalImport?  that's what I was looking for.

btw I agree this case is invalid but am don't think your current code catches it.